connector/extensions/web-service/src/test/java/org/oaebudt/edc/web/service/ReportServiceTest.java

```diff
@@ -0,0 +1,137 @@
+package org.oaebudt.edc.web.service;
+
+import jakarta.json.JsonException;
+import org.eclipse.edc.connector.controlplane.asset.spi.domain.Asset;
+import org.eclipse.edc.connector.controlplane.contract.spi.types.offer.ContractDefinition;
+import org.eclipse.edc.connector.controlplane.services.spi.asset.AssetService;
+import org.eclipse.edc.connector.controlplane.services.spi.contractdefinition.ContractDefinitionService;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.result.ServiceResult;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.oaebudt.edc.web.model.ReportType;
+import org.oaebudt.edc.web.repository.ReportStore;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ReportServiceTest {
+
+    private ReportService reportService;
+    private ReportStore reportStore;
+    private Monitor monitor;
+    private AssetService assetService;
+    private ContractDefinitionService contractDefinitionService;
+    private URI consumerApiBaseUrl;
+
+    private static Stream<Arguments> invalidMetadataProvider() {
+        return Stream.of(
+                Arguments.of("{}", "legalOrganizationName is required"),
+                Arguments.of("{\"legalOrganizationName\":\"\",\"countryOfOrganization\":\"US\",\"contactPerson\":\"Jane\",\"dataProcessingDescription\":\"desc\",\"qualityAssuranceMeasures\":\"qa\",\"dataLicensingTerms\":\"terms\",\"organizationWebsite\":\"http://valid.com\",\"contactEmail\":\"valid@example.com\",\"dataAccuracyLevel\":1,\"dataGenerationTransparencyLevel\":1,\"dataDeliveryReliabilityLevel\":1,\"dataFrequencyLevel\":1,\"dataGranularityLevel\":1,\"dataConsistencyLevel\":1}", "legalOrganizationName is required"),
```

Comment:
I'd suggest to use the `Json` class to create and serialize jsons, or jackson at least, because manually created jsons are a pain to read and maintain.
Same for all the other manual json creation.

Reply:
Resolved